### PR TITLE
Creates Index.integration.test.jsx and Sketchlist.unit.test.jsx

### PR DESCRIPTION
**Pull request by khanniie**
Creates a few new tests, tweaks some testing helper files to work for those tests, add dev libraries msw and redux-mock-store.

Tests added:
- Sketchlist test
- Projects action creator test
- index.jsx integration test
- very barebones Editor.unit.test.jsx test

Other files added:
- Stylemock to mock out style files
- i18n.js mock to fix ERRCONNECTED problem
- test data

* [x] has no linting errors (`npm run lint`)
* [x] is from a uniquely-named feature branch and has been rebased on top of the latest `develop` branch. (If I was asked to make more changes, I have made sure to rebase onto `develop` then too)
* [x] is descriptively named and links to an issue number, i.e. `Fixes #123`

### CODE REVIEW

File: developer_docs/testing.md [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+
+## Testing dependencies
+1. [Jest](https://jestjs.io/)
+2. [react-testing-library](https://testing-library.com/docs/react-testing-library/intro/)
+3. [redux-mock-store](https://github.com/reduxjs/redux-mock-store)
+
+## Useful testing commands
+Run the whole test suite
+```
+npm run test
```
Comment (catarak) [inline developer_docs/testing.md]: Usually for shell commands in documentation I preface them with a `$`

File: developer_docs/testing.md [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+
+### Troubleshooting
+1. Check if the component makes any API calls. If it's using axios, jest should already be set up to replace the axios library with a mocked version; however, you may want to [mock](https://jestjs.io/docs/mock-function-api#mockfnmockimplementationoncefn) the axios.get() function with your own version so that GET calls "return" whatever data makes sense for that test. 
+
+    ```
+    axios.get.mockImplementationOnce(
+      (x) => Promise.resolve({ data: 'foo' })
+    );
+    ```
+You can see it used in the context of a test [here](../client/modules/IDE/components/SketchList.test.jsx).
```
Comment (catarak) [inline developer_docs/testing.md]: For screen reader accessibility reasons, it's best practice to not have links with the text "here" or  "click here", but to instead describe what the link goes to. See [What's wrong with using "Click Here" links?](https://webaccess.berkeley.edu/ask-pecan/click-here)

File: developer_docs/testing.md [hunk 3]
```diff
@@ -0,0 +1,472 @@
...
+(the below assumes we're using proposed folder structure 1)
+1. Make a new file in the ``__tests__`` folder that's directly adjacent to your file. For example, if ``example.jsx`` is in ``src/components``, then you would make a file called ``example.test.jsx`` in ``src/components/__tests__``
+2. Check if the component is connected to redux or not.
+3. If it is, see the redux section below on how to write tests for that.
+4. If it's not, see the section below on writing tests for unconnected components.
+
+### What to test
+For any type of component, you might want to consider testing:
+- User input results in the class's method being called. 
+  ```
```
Comment (catarak) [inline developer_docs/testing.md]: if you change this line to "```js", then this code will be syntax highlighted. same with all of the other instances of code. See [Creating and highlighting code blocks](https://docs.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks).

**Review by catarak — COMMENTED** [review]
Just some formatting suggestions!

File: developer_docs/testing.md (hunk 1 above)
Comment (khanniie) [reply developer_docs/testing.md]: fixed :-) thank you!

File: developer_docs/testing.md (hunk 2 above)
Comment (khanniie) [reply developer_docs/testing.md]: fixed, good to know! Thanks for letting me know~

File: developer_docs/testing.md (hunk 3 above)
Comment (khanniie) [reply developer_docs/testing.md]: wow, MAGIC! It worked and looks much better :))

**Review by catarak — APPROVED** [review]

**Pull request merged**